### ISSUE TITLE
Implement Batch-Normalization

### ISSUE THREAD

**Issue by lucasb-eyer** [lucasb-eyer/lbtoolbox#1]
http://arxiv.org/abs/1502.03167

**Comment by lucasb-eyer** [comment]
So turns out this also makes a difference between training-time and testing-time, just like Dropout #4. Guess I'll have to refactor a bit so I can easily implement both of them.

**Comment by lucasb-eyer** [comment]
Moved to lucasb-eyer/DeepFried#1.
